twa

```diff
@@ -467,44 +467,49 @@ function stage_5_robots_check {
 function stage_6_caa {
   verbose "Stage 6: CAA checks"
 
-  records=$(dig +short caa "${domain}")
+  records=$(dig +noall +answer caa "${domain}")
 
   issuers=()
   wildcard_issuers=()
   iodefs=()
 
   if [[ -n "${records}" ]]; then
-    while read -r flag tag value; do
-      value="${value//\"}"
-
-      if [[ -n "${flag}" && -n "${tag}" && -n "${value}" ]]; then
-        if [[ "${flag}" -eq 0 ]]; then
-          if [[ "${tag}" == "issue" ]]; then
-            if [[ -n "${value}" ]]; then
-              issuers+=("${value}")
+    while read -r -a RECORD; do
```

Comment:
Instead of reading an array and then breaking the individual variables out, maybe just read them explicitly?

```bash
while read -r type flag tag value; do
# ...
done
```

`value` then needs to be unquoted like before, but I think that's a bit cleaner overall :slightly_smiling_face:

Reply:
I had first pushed it with reading the variables explicitly like that, but shellcheck didn't like that not all of them were used and failed CI so I committed an update.

Reply:
I should say, I agree that it's not as clean.  But I don't know a great way to skip the first N tokens when `read`-ing into variables like that.

Reply:
Ah, my bad -- I didn't realize that they weren't starting at `0`. In that case, my only nit is to make `RECORD` lowercase for consistency: either `record` or `caa_record` would be fine.

Reply:
I went with `record` because I find not all lines of output are actually caa records.

Reply:
Good call.